api/client/catalog/catalog.go

```diff
@@ -43,3 +57,47 @@ func IsCatalogAlreadyExistsError(err error) bool {
 	}
 	return apiErr.HTTPStatusCode == 400 && strings.Contains(apiErr.Message, "already exists")
 }
+
+// getAllResourcesWithPagination is a generic helper function that fetches all items from a paginated catalog API endpoint
+func getAllResourcesWithPagination[T any](
```

Comment:
```suggestion
func getAllPaginated[T any](
```

nit: keeping the name of function small and focussed.